dex/fiatrates/sources.go

```diff
@@ -0,0 +1,246 @@
+					fiatRates[parseSymbol(sym)] = rate
+				}
+
+				return fiatRates, nil
+			},
+		},
+		{
+			name:            binance,
+			requestInterval: defaultRefreshInterval,
+			disabled:        strings.Contains(disabledSources, strings.ToLower(binance)),
+			getRates: func(ctx context.Context, assets []string, _ dex.Logger) (map[string]float64, error) {
+				priceEndpoint := binancePriceEndpoint
+				if cfg.EnableBinanceUS {
+					priceEndpoint = binanceUSPriceEndpoint
+				}
+
+				symbols := parseBinanceSymbols(assets)
+				if symbols == "" {
+					return nil, nil // nothing to fetch
+				}
+
+				var response []*struct {
+					Symbol string `json:"symbol"`
+					Price  string `json:"price"`
+				}
+
+				reqURL := fmt.Sprintf(priceEndpoint, url.PathEscape(symbols))
+				err := getRates(ctx, reqURL, &response)
+				if err != nil {
+					return nil, fmt.Errorf("unable to fetch fiat rates: %w", err)
+				}
+
+				fiatRates := make(map[string]float64)
+				for _, asset := range response {
+					symbol := parseSymbol(strings.TrimSuffix(asset.Symbol, "USDT"))
+					rate, err := strconv.ParseFloat(asset.Price, 64)
+					if err != nil {
+						log.Error("%s: failed to convert fiat rate for %s to float64: %v", binance, symbol, err)
+						continue
+					}
+					fiatRates[symbol] = rate
+				}
+
+				return fiatRates, nil
+			},
+		},
+		{
+			name:            coinpaprika,
+			requestInterval: defaultRefreshInterval,
+			disabled:        strings.Contains(disabledSources, strings.ToLower(coinpaprika)),
+			getRates: func(ctx context.Context, assets []string, log dex.Logger) (map[string]float64, error) {
+				fiatRates := make(map[string]float64, len(assets))
+				for _, a := range assets {
+					fiatRates[parseSymbol(a)] = 0
+				}
+
+				var res []*struct {
+					Symbol string `json:"symbol"`
+					Quotes struct {
+						USD struct {
+							Price float64 `json:"price"`
+						} `json:"USD"`
+					} `json:"quotes"`
+				}
+
+				if err := getRates(ctx, coinpaprikaPriceEndpoint, &res); err != nil {
+					return nil, err
+				}
+
+				for _, coinInfo := range res {
+					symbol := parseSymbol(coinInfo.Symbol)
+					_, found := fiatRates[symbol]
+					if !found {
+						continue
+					}
+
+					price := coinInfo.Quotes.USD.Price
+					if price == 0 {
+						log.Errorf("zero-price returned from coinpaprika for asset with symbol %s", symbol)
+						continue
+					}
+
+					fiatRates[symbol] = price
+				}
+
+				return fiatRates, nil
+			},
+		},
+		{
+			name:            messari,
+			requestInterval: messariRefreshInterval,
+			disabled:        strings.Contains(disabledSources, strings.ToLower(messari)),
+			getRates: func(ctx context.Context, assets []string, log dex.Logger) (map[string]float64, error) {
+				fiatRates := make(map[string]float64)
+				for _, sym := range assets {
+					res := new(struct {
+						Data struct {
+							MarketData struct {
+								Price float64 `json:"price_usd"`
+							} `json:"market_data"`
+						} `json:"data"`
+					})
```

Comment:
You're kinda all over the place with how you're defining these response structs. Prefer this pattern please.

```Go
var res struct {
	Data struct {
		MarketData struct {
			Price float64 `json:"price_usd"`
		} `json:"market_data"`
	} `json:"data"`
)
doThing(&res)
```